src/hotkey/hotkey_item.cpp

```diff
@@ -432,13 +442,15 @@ void reset_default_hotkeys()
 	}
 }
 
-const hotkey_list& get_hotkeys()
+const threadsafe_hotkey_list get_hotkeys()
 {
-	return hotkeys_;
+	return threadsafe_hotkey_list(hotkeys_, hotkeys_mutex_);
 }
 
 void save_hotkeys(config& cfg)
 {
+	lock_guard<mutex> lock(hotkeys_mutex_);
```

Comment:
If this lock is necessary then there's likely to be a preference-corrupting bug somewhere. The existence of this lock implies that somewhere there's code which modifies the list, and which might try to run at the same time - adding the lock removes one of these 3 possibilities, but leaves the other two possible:

* modification happens, then list is saved to `cfg`
* list is saved to `cfg`, then modification happens
* (prevented by lock) modification happens while this function is iterating over `hotkeys_`

Reply:
The problem mentioned in `game_config_manager` indicated that it was possible an event could modify the hotkey list at the same time as `load_hotkeys()` was being called by the worker thread running `game_config_manager::init_game_config`.

load_hotkeys() can both erase and push_back to the hotkey list, both of which can invalidate the iterator here.

I don't really know how to - or if it's possible to - ensure which things can definitely never be called while another thread is messing with the hotkey list. So i just slapped a mutex on everything that touched it...

Reply:
Is it possible for `load_hotkeys` to load into a local list and just swap when done? Might that be simpler?

Reply:
I actually think it would be better to do that (swapping in stead of directly editing everything in a worker thread) for everything in those threads. But the problem again there is that if something modifies the hotkey list while it is being loaded, then the swap will clobber any changes during that time.

For all of this... i honestly have no idea why anything would ever want to be modifying hotkeys via event while the loading screen is displaying. All i have to go on is the fact that there is an existing block there. Perhaps clobbering is fine? I have no idea.